### DISCUSSION TITLE
How the 24-hour execution mechanism works in the contract?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#408] [Solidity / smart contracts] [answered]
How the 24-hour execution mechanism works in the contract?

**Comment by solimander** [comment, marked as answer]
When the auction is kicked off, the auction's `startTime` is set to the `block.timestamp` and its `endTime` is set to startTime + duration. See [here](https://github.com/nounsDAO/nouns-monorepo/blob/04c96bd35e9dcebd983db37dfd59248b0c90284a/packages/nouns-contracts/contracts/NounsAuctionHouse.sol#L197).

Once the current `block.timestamp` is greater than or equal to the `endTime`, the auction is considered to have ended. At this time, you cannot place a bid anymore. You can only settle the auction, which subsequently kicks off the next auction. See [here](https://github.com/nounsDAO/nouns-monorepo/blob/04c96bd35e9dcebd983db37dfd59248b0c90284a/packages/nouns-contracts/contracts/NounsAuctionHouse.sol#L108).
